alibi/explainers/anchor_tabular.py

```diff
@@ -958,19 +985,50 @@ def add_names_to_exp(self, explanation: dict) -> None:
                 handled.add(feat_id)
             explanation['names'].append(fname)
 
+    @property
+    def predictor(self) -> Callable:
+        if self.ohe:
+            # return the predictor that expect one-hot encoded categorical features
+            return self._ohe_predictor
+
+        # return the encoder that expects ordinal/label encoded categorical features
+        return self._predictor
+
     def _transform_predictor(self, predictor: Callable) -> Callable:
-        # check if predictor returns predicted class or prediction probabilities for each class
-        # if needed adjust predictor so it returns the predicted class
-        if np.argmax(predictor(np.zeros([1, len(self.feature_names)])).shape) == 0:
+        try:
+            # check if predictor returns predicted class or prediction probabilities for each class
+            # if needed adjust predictor so it returns the predicted class
+            prediction = predictor(np.zeros([1, len(self.feature_names)]))
+        except Exception:
+            raise ValueError("The classifier is expecting a different number of features.")
```

Comment:
We should be more explicit in the error message, e.g. add "Check the values of `feature_names` and `categorical_names`", or even, if it's possible to keep track of the inferred number of columns, "Calling the classifier using a datapoint with {n_cols} failed, check the values of `feature_names` and `categorical_names`."